spec/controllers/admin/order_cycles_controller_spec.rb

```diff
@@ -337,5 +337,31 @@ module Admin
         end
       end
     end
+
+    describe "#subscriptions_counts" do
+      let(:oc1) { create(:simple_order_cycle) }
+      let(:oc2) { create(:simple_order_cycle) }
+      let(:collection) { OrderCycle.where(id: [oc1, oc2]) }
+
+      context "when the collection has not been set" do
+        it "returns and empty array" do
+          expect(controller.send(:subscriptions_counts)).to eq []
+        end
+      end
+
+      context "when the collection has been set" do
+        let!(:po1) { create(:proxy_order, order_cycle: oc1) }
+        let!(:po2) { create(:proxy_order, order_cycle: oc1) }
+        let!(:po3) { create(:proxy_order, order_cycle: oc2) }
+
+        before { controller.instance_variable_set(:@collection, collection) }
```

Comment:
IMO we can't afford to do this. As it violates the controller's encapsulation it will break sooner than later. 

this, the need to use `#send` and the fact that we add yet another method that is not directly related to the controller's responsibilities leads me to think that we should rather extract `subscriptions_count` as a `SubscriptionsCount` plain ruby object. This would act as a query object. https://robots.thoughtbot.com/active-record-eager-loading-with-query-objects-and-decorators is good article on the topic.

Reply:
@sauloperez Ok, see latest commit.